cheevos/cheevos.c

```diff
@@ -3441,32 +3441,46 @@ CORO_again: ;
 
                   if (cheevos_locals.token[0])
                      CORO_RET();
+                  bool via_token = false;
 
                   {
                      char urle_user[64];
-                     char urle_pwd[64];
+                     char urle_login[64];
                      const char *username = coro ? coro->settings->arrays.cheevos_username : NULL;
-                     const char *password = coro ? coro->settings->arrays.cheevos_password : NULL;
+                     const char *login;
 
-                     if (!username || !*username || !password || !*password)
+                     if (coro)
+                     {
+                        if (coro->settings->arrays.cheevos_token[0])
```

Comment:
I think it would be better to check here also for ``*coro->settings->arrays.cheevos_password == '\0'`` .

That way you can generate a new token by entering your password again. Also I guess it's good to prevents people from needlessly keeping the password stored in the config file when they have already a token.